### ISSUE TITLE
Send data to different Kinesis stream

### ISSUE THREAD

**Issue by jenishshah671993** [awslabs/aws-fluent-plugin-kinesis#165]
Hi,
I have JSON data. On the basis of JSON key value I need to send data to different kinesis stream. So can you suggest me if it's possible

**Comment by cosmo0920** [comment]
It is possible to implement with built-in placeholders, which is the part of brand new Fluentd v1 API, for https://github.com/awslabs/aws-fluent-plugin-kinesis/blob/master/lib/fluent/plugin/out_kinesis_streams.rb#L49
Currently, it is not implemented.

**Comment by cosmo0920** [comment]
I've registered [a PR](https://github.com/awslabs/aws-fluent-plugin-kinesis/pull/174) to fix this issue.
Does anyone want to try out this?

**Comment by simukappu** [comment]
Thank you for your contributing.
I think we need to separate buffer for each destination streams for plugin durability. What do you think?

**Comment by cosmo0920** [comment]
@simukappu Fluentd v1 separates buffer for each destination streams:
https://docs.fluentd.org/v1.0/articles/buffer-section#other-keys

With [the testing configuration](https://github.com/awslabs/aws-fluent-plugin-kinesis/pull/174#issuecomment-466851293), Fluentd handles buffer per `$.key.nested`.

**Comment by simukappu** [comment]
Sounds good. Does anybody have any feedback for this issue or PR?

**Comment by amitdhawan** [comment]
Is this done?
I want to use this feature.

**Comment by amitdhawan** [comment]
> Sounds good. Does anybody have any feedback for this issue or PR?

Lets merge it.

**Comment by cosmo0920** [comment]
@amitdhawan Can #174 patch work under your created AWS kinesis environment?
Testing on a few users are required.

**Comment by amitdhawan** [comment]
Should i install this plugin from this branch on my fluentd config? How can i do that, could you please share the steps.I can test that on my AWs environment then.

**Comment by amitdhawan** [comment]
Let me know the steps how can install this plugin on Fluentd. I can test it on my environment then.

**Comment by cosmo0920** [comment]
```
$ git clone git@github.com:awslabs/aws-fluent-plugin-kinesis.git
$ git fetch origin pull/174/head:pr-174
$ git checkout pr-174
$ gem build fluent-plugin-kinesis.gemspec
```

And then, 

```
$ td-agent-gem install fluent-plugin-kinesis-*.gem
# Or
$ fluent-gem install fluent-plugin-kinesis-*.gem
```

**Comment by amitdhawan** [comment]
ok..let me give it a shot.

**Comment by simukappu** [comment]
Released as v3.2.0.
https://rubygems.org/gems/fluent-plugin-kinesis/versions/3.2.0

If you have any feedback, feel free to share with us.
